index.d.ts

```diff
@@ -10,8 +10,27 @@ export interface MonacoTailwindcssOptions {
   tailwindConfig?: TailwindConfig;
 }
 
+/**
+ * This type represents either a string value, a model whose value will be retrieved, or a URI which
+ * refers to the model whose value to get.
+ */
+export type ModelLike = editor.ITextModel | Uri | string;
+
 export interface MonacoTailwindcss extends IDisposable {
   setTailwindConfig: (tailwindConfig: TailwindConfig) => void;
+
+  /**
+   * Generate styles using Tailwindcss.
+   *
+   * This generates CSS using the Tailwind JIT compiler. It uses the Tailwind configuration that has
+   * previously been passed to {@link configureMonacoTailwindcss}.
+   *
+   * @param css - The CSS to process. Only one CSS file can be processed at a time.
+   * @param content - All content that contains CSS classes to extract.
+   * @returns The CSS generated by the Tailwind JIT compiler. It has been optimized for the given
+   * content.
+   */
+  generateStylesFromContent: (css: ModelLike, content: ModelLike[]) => Promise<string>;
```

Comment:
As can be seen here, this method does **not** use the internal Monaco models. This allows the user to explicitly specify which values to include.

While typing this, I considered it might be nice to add an `@example` annotation here. WDYT?

```js
monacoTailwindcss.generateStylesFromContent(
  `
  @tailwind base;
  @tailwind components;
  @tailwind utilities;
  `,
  editor.getModels().filter((model) => model.getLanguageId() === 'html')
)
```

Reply:
this is great! Idk if jsdocs likes `@` in the examples but this looks good ;)

Reply:
`ModelLike` seemed wierd at first but this is easier to use ^^

Reply:
`eslint-plugin-jsdocs` doesn’t seem to like it, so I just used an identifier named `css`.

It seemed nice to me to be able to just pass a model, but maybe it’s a bit too _magic_ then?

Internally Tailwind allows to use a different extractor based on the file type, but so far it seems it always extracts the entire class names known from the content.

Reply:
this is super fine for know but when the tw behavior changes based on file extension we should provide that as option too for direct passed string inputs

Reply:
or we directly allow only ChangedContent[]

Reply:
Directly passing `ChangedContent[]` seems good too. Should we still support strings, models, and URIs? I kind of like the idea of passing strings or models, as our users will be dealing with that anyway. I’m not so sure about URIs.

Reply:
hmm we need to limit the polymorphism a bit ^^
i get your point, but when using this library this function will not be called in that many different places so i would be fine also with only allowing `ChangedContent[]`. Less polymorphism makes the api easier to maintain.

This is just my opinion. Feel free to build it as you like i dont want to block this pr.

...`URI` is the least important thing, we should probably drop it when whe continue with polymorphism. But `ChangedContent[]` is a must as i find.

Reply:
I updated it to accept `ChangedContent | string` instead, except I named `ChangedContent` `Content`. I think the term `ChangedContent` makes sense in Tailwind internally, but in this case I think it’s a bit weird user facing, as nothing has changed.